### ISSUE TITLE
Update the `transformers` integration

### ISSUE THREAD

**Issue by rlouf** [outlines-dev/outlines#806]
In the vein of #782 and #772 we should refactor the `transformers` integration to use logits processor.

We will keep the custom sampling loop, but via a `Outlines` model that wraps `transformers` models. We should be able to remove `torch` and `transformers` as default dependencies as a result.
